master/internal/user/authz_rbac.go

```diff
@@ -243,37 +244,62 @@ func (a *UserAuthZRBAC) CanCreateAccessToken(
 	return nil
 }
 
-// CanGetAccessTokens returns an error if the user does not have permission to view either
-// their own token or another user's token based on the targetUser.
-func (a *UserAuthZRBAC) CanGetAccessTokens(
-	ctx context.Context, curUser, targetUser model.User,
-) (err error) {
+func canAdministrateAccessTokens(ctx context.Context, curUserID model.UserID) (err error) {
 	fields := audit.ExtractLogFields(ctx)
-
-	// TODO: improve logging around the case were a user is viewing their own token's description
-	if curUser.ID == targetUser.ID {
-		err = db.DoesPermissionMatch(ctx, curUser.ID, nil,
-			rbacv1.PermissionType_PERMISSION_TYPE_VIEW_TOKEN)
-		if err != nil {
-			return errors.Wrap(err, "unable to get token due to insufficient permissions")
-		}
-		return nil
-	}
-
-	logCanAdministrateAccessTokenOnUser(fields, curUser.ID,
+	logCanAdministrateAccessTokenOnUser(fields, curUserID,
 		rbacv1.PermissionType_PERMISSION_TYPE_VIEW_OTHER_TOKEN)
 	defer func() {
 		audit.LogFromErr(fields, err)
 	}()
 
-	err = db.DoesPermissionMatch(ctx, curUser.ID, nil,
+	return db.DoesPermissionMatch(ctx, curUserID, nil,
 		rbacv1.PermissionType_PERMISSION_TYPE_VIEW_OTHER_TOKEN)
-	if err != nil && curUser.ID != targetUser.ID {
+}
+
+// CanGetAccessTokens returns an error if the user does not have permission to view own or
+// another user's token based on own role permissions.
+func (a *UserAuthZRBAC) CanGetAccessTokens(
+	ctx context.Context, curUser model.User, query *bun.SelectQuery, filterUserID model.UserID,
```

Comment:
nit (non-blocking): I'm concerned `filterUserID` might not be as clear as the `targetUserID` or `accessTokenUserID` could be here

Reply:
Makes sense. Will change it.